sparse/unit_test/Test_Sparse_extractCrsDiagonalBlocks.hpp

```diff
@@ -0,0 +1,156 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include "KokkosSparse_Utils.hpp"
+#include "KokkosKernels_TestUtils.hpp"
+
+namespace Test {
+template <typename scalar_t, typename lno_t, typename size_type,
+          typename device>
+void run_test_extract_diagonal_blocks(int nrows, int nblocks) {
+  using RowMapType     = Kokkos::View<size_type *, device>;
+  using EntriesType    = Kokkos::View<lno_t *, device>;
+  using ValuesType     = Kokkos::View<scalar_t *, device>;
+  using RowMapType_hm  = typename RowMapType::HostMirror;
+  using EntriesType_hm = typename EntriesType::HostMirror;
+  using ValuesType_hm  = typename ValuesType::HostMirror;
+  using crsMat_t       = CrsMatrix<scalar_t, lno_t, device, void, size_type>;
+
+  crsMat_t A;
+  std::vector<crsMat_t> DiagBlks(nblocks);
+
+  if (nrows != 0) {
+    // Generate test matrix
+    const size_type nnz = 2 + (nrows - 2) * 3 + 2;
```

Comment:
I would suggest making this a random number modulo some upper bound. Is there a particular reason for using the constants 2 and 3 in this expression>

Reply:
@e10harvey For this test, I just want to have a simple diagonal matrix with one main diagonal and two diagonals: above and below the main one. So, the first row and the last row have 2 non-zero elements and the other rows have 3 non-zero elements.